pkg/authentication/example_test.go

```diff
@@ -69,3 +72,146 @@ func ExampleApiClient_Octocat() {
+	// this responseHandler function panics:
+	// panic: interface conversion: *http.Response is not serialization.Parsable: missing method GetFieldDeserializers
+	// not sure if i'm mishandling something in this totally non-implemented function or what
+	responseHandler := func(response interface{}, errorMappings abstractions.ErrorMappings) (interface{}, error) {
+		// handle rate-limited responses here
+		fmt.Printf("%v\n", response)
+		if response == nil {
+			return nil, nil
+		}
+		return response, nil
+	}
+	responseHandlerOption := abstractions.NewRequestHandlerOption()
+	responseHandlerOption.SetResponseHandler(responseHandler)
+
+	options := make([]abstractions.RequestOption, 0)
+	options = append(options, responseHandlerOption)
+
+	retryHandlerOption := &kiotaHttp.RetryHandlerOptions{
+		MaxRetries: 3,
+	}
+	options = append(options, retryHandlerOption)
+
+	tokenProvider := auth.NewTokenProvider(
+		// to create an authenticated provider, uncomment the below line and pass in your token
+		// auth.WithAuthorizationToken("ghp_your_token"),
+		auth.WithUserAgent("octokit/go-sdk.example-functions"),
+	)
+
+	adapter, err := kiotaHttp.NewNetHttpRequestAdapter(tokenProvider)
+	if err != nil {
+		log.Fatalf("Error creating request adapter: %v", err)
+	}
+	client := github.NewApiClient(adapter)
+
+	s := "Salutations"
+
+	// create headers that accept json back; GitHub's OpenAPI definition says
+	// octet-stream but that's not actually what the API returns in this case
+	headers := abstractions.NewRequestHeaders()
+	_ = headers.TryAdd("Accept", "application/vnd.github.v3+json")
+
+	octocatRequestConfig := &abstractions.RequestConfiguration[octocat.OctocatRequestBuilderGetQueryParameters]{
+		QueryParameters: &octocat.OctocatRequestBuilderGetQueryParameters{
+			S: &s,
+		},
+		Headers: headers,
+		Options: options,
+	}
+
+	cat, err := client.Octocat().Get(context.Background(), octocatRequestConfig)
+	if err != nil {
+		log.Fatalf("error getting octocat: %v", err)
+	}
+	fmt.Printf("%v\n", string(cat))
+	// Output:
+	// MMM.           .MMM
+	//                MMMMMMMMMMMMMMMMMMM
+	//                MMMMMMMMMMMMMMMMMMM      _____________
+	//               MMMMMMMMMMMMMMMMMMMMM    |             |
+	//              MMMMMMMMMMMMMMMMMMMMMMM   | Salutations |
+	//             MMMMMMMMMMMMMMMMMMMMMMMM   |_   _________|
+	//             MMMM::- -:::::::- -::MMMM    |/
+	//              MM~:~ 00~:::::~ 00~:~MM
+	//         .. MMMMM::.00:::+:::.00::MMMMM ..
+	//               .MM::::: ._. :::::MM.
+	//                  MMMM;:::::;MMMM
+	//           -MM        MMMMMMM
+	//           ^  M+     MMMMMMMMM
+	//               MMMMMMM MM MM MM
+	//                    MM MM MM MM
+	//                    MM MM MM MM
+	//                 .~~MM~MM~MM~MM~~.
+	//              ~~~~MM:~MM~~~MM~:MM~~~~
+	//             ~~~~~~==~==~~~==~==~~~~~~
+	//              ~~~~~~==~==~==~==~~~~~~
+	//                  :~==~==~==~==~~
+}
+
+func Example_HttpClientForRateLimiting() {
```

Comment:
This function (currently with intentional malformed example suffix) uses a Go standard library HTTP client with a RoundTripper with logic mostly cribbed from `integrations/terraform-provider-github` for rate limiting. It does not make use of the Kiota helpers.